coordinator/migrations/2023-07-11-104904_create_channels/up.sql

```diff
@@ -0,0 +1,22 @@
+-- Your SQL goes here
+CREATE TYPE "ChannelState_Type" AS ENUM (
+    'Pending',
+    'Open',
+    'Closed',
+    'ForceClosedRemote',
+    'ForceClosedLocal'
+);
+CREATE TABLE "channels" (
+    user_channel_id TEXT PRIMARY KEY,
+    channel_id TEXT UNIQUE,
+    capacity BIGINT NOT NULL,
+    balance BIGINT NOT NULL,
+    funding_txid TEXT,
+    channel_state "ChannelState_Type" NOT NULL,
+    counterparty_pubkey TEXT NOT NULL,
+    created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    updated_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    -- this value is stored here for simplicity of creating a sql query. However, it is not the source of truth and can
+    -- be recreated from the various transactions attached to the channel.
+    costs BIGINT NOT NULL DEFAULT 0
```

Comment:
🙃 What do we consider "costs" in the context of this table - is this the on-chain transaction fees (only)? Maybe some more words what these "costs" actually represent would be nice.

Reply:
Another thought on this.
I started working on https://github.com/get10101/10101/issues/577 and it seems logical to add a `opening_fee_payment_hash` to the `channels` table that would record the `PaymentHash` that associates the JIT channel opening payment with the channel. 
I would not reflect that in the `costs` field though which made me thing if we can find a better name for `costs` - maybe `aggregated_transaction_costs` or similar?

Reply:
After an (unrelated) chat with @luckysori I came to the conclusion that it would be better to track the transaction costs based on the published transactions rather than this `cost` field.

We recently agreed that we should track all broadcasted transactions - see https://github.com/get10101/10101/issues/954 - if we do this first we can actually associate the transactions with the channel opening.
I think this is the way to go :)

Here's how this could look:

```sql
CREATE TABLE "channels" (
    user_channel_id TEXT PRIMARY KEY,
    channel_id TEXT UNIQUE,
    capacity BIGINT NOT NULL,
    balance BIGINT NOT NULL,
    funding_txid TEXT REFERENCES transactions(txid),
    commit_txid TEXT REFERENCES transactions(txid),
    closing_txid TEXT REFERENCES transactions(txid),
    channel_state "ChannelState_Type" NOT NULL,
    counterparty_pubkey TEXT NOT NULL,
    created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
    updated_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
);

-- All the broadcasted transactions
CREATE TABLE transactions (
    txid TEXT PRIMARY KEY
    fee BIGINT NOT NULL DEFAULT 0,
    -- either our or counterparty pubkey
    broadcasted_by: TEXT NOT NULL,
);
```

That way we keep the associations clear:

- On-chain costs must be associated with a transaction
- Off-chain costs (Lightning channel related) must be associated with a payment
- (Trade pnl is virtual and handled outside of payments and transactions)

Reply:
I like the idea of storing the transaction separately. That allows us to associate fees directly to the transaction. 👍